### PR TITLE
Updated several pods and locked more to a specific version

### LINKED ISSUE #7569: Version lock several pods, update others
There are several cases of the Podfile containing loose/fuzzy version numbers. Version lock as many pods as possible and bring others current.

### PR DESCRIPTION
**Pull request by astralbodies**
Fixes #7569 

As part of our release process we try to update the CocoaPods dependencies once in a while. I noticed several pods with version numbers not fixed at a specific version number. This corrects that as much as possible.

To test:
1. Clean, build, and run the unit tests.
2. Make sure notifications sync still works while in the app (StarScream updated).

StarScream was the only major pod revision update with this (look at the lock file).

Needs review: @elibud, @koke, @jleandroperez, @nheagy

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -8,10 +8,10 @@ workspace 'WordPress.xcworkspace'
 
 ## Pods shared between all the targets
 def shared_with_all_pods
-  pod 'CocoaLumberjack', '~> 3.2.0'
-  pod 'FormatterKit/TimeIntervalFormatter', '~> 1.8.1'
+  pod 'CocoaLumberjack', '3.2.0'
+  pod 'FormatterKit/TimeIntervalFormatter', '1.8.2'
   pod 'NSObject-SafeExpectations', '0.0.2'
-  pod 'UIDeviceIdentifier', '~> 0.1'
+  pod 'UIDeviceIdentifier', '~> 0.4'
```
Comment (koke) [inline Podfile]: The newest version is 1.1.1, this is [missing](https://github.com/squarefrog/UIDeviceIdentifier/compare/0.5.0...1.1.1) several iPhone/iPad models. The API hasn't changed, so updating to the latest should be fine. Although I think we could just drop this one and use the raw platform string in logs.

**Comment by koke** [comment]
> Do we have an automated way to do it?

There's `pod outdated`:

```
$ pod outdated
[...]
Analyzing dependencies
The following pod updates are available:
- Alamofire 4.0.1 -> 4.0.1 (latest version 4.5.0)
- Crashlytics 3.8.4 -> 3.8.5 (latest version 3.8.5)
- Expecta 1.0.5 -> 1.0.5 (latest version 1.0.6)
- Fabric 1.6.11 -> 1.6.12 (latest version 1.6.12)
- HockeySDK 4.1.5 -> 4.1.6 (latest version 4.1.6)
- lottie-ios 1.5.1 -> 1.5.2 (latest version 1.5.2)
- MRProgress 0.7.0 -> 0.7.0 (latest version 0.8.3)
- Specta 1.0.5 -> 1.0.5 (latest version 1.0.6)
- Starscream 2.0.2 -> 2.0.2 (latest version 2.1.0)
- UIDeviceIdentifier 0.5.0 -> 0.5.0 (latest version 1.1.1)
- WordPress-Aztec-iOS 1.0.0-beta.7 -> 1.0.0-beta.7 (latest version 1.0.0-beta.8)
```

File: Podfile (hunk 1 above)
Comment (astralbodies) [inline Podfile]: Automattic-Tracks-iOS has a pod dependency of `pod 'UIDeviceIdentifier', '~> 0.4'` so I can't go any higher than this without updating that podspec (which I eventually will do).

**Comment by astralbodies** [comment]
@koke 

> I don't understand the logic behind pinning to specific versions instead of using ~>, unless you know an upgrade would break something in particular.

The only reason we pin versions is so that upgrading to new pod versions is not an implicit process - someone has to intentionally update the versions and review any changes incurred. Having the fuzzy version numbers allows someone to sneak in an accidental `pod update` in a PR with unintended version bumps in the lock file.

The other reality is I do not trust the maintainers of the pods to follow semantic versioning methodologies. I've encountered breaking changes before in a patch level change (1.1.x).

**Comment by astralbodies** [comment]
@elibud 

> What's the criteria to decide which ones are locked and which ones aren't?

The ones that didn't get locked (like UIDeviceInformation) have a pod dependency on them and I can't get tighter on the versioning. Automattic-Tracks-iOS lists UIDeviceInformation at `~>0.4` so that's the tightest I can go.

**Comment by koke** [comment]
> that upgrading to new pod versions is not an implicit process - someone has to intentionally update the versions and review any changes incurred

I'd argue that running `pod update` is already explicit. If you want to update a single pod, you can do `pod update SomeLibrary` and leave the rest unchanged. You shouldn't just run `pod update` if you don't want to update all dependencies.

I'm not too bothered by this, but I think it makes it harder to actually update all dependencies when we want to do that. But then, it's probably a good idea to review the changelogs anyway.

**Pull request merged**